sway/layout.c

```diff
@@ -432,9 +432,29 @@ static void arrange_windows_r(swayc_t *container, double width, double height) {
 		{
 			struct wlc_size resolution = *wlc_output_get_resolution(container->handle);
 			width = resolution.w; height = resolution.h;
+			// output must have correct size due to e.g. seamless mouse,
+			// but a workspace might be smaller depending on panels.
+			container->width = width;
+			container->height = height;
+		}
+		// arrange all workspaces:
+		for (i = 0; i < container->children->length; ++i) {
+			swayc_t *child = container->children->items[i];
+			arrange_windows_r(child, -1, -1);
+		}
+		// Bring all unmanaged views to the front
+		for (i = 0; i < container->unmanaged->length; ++i) {
+			wlc_handle *handle = container->unmanaged->items[i];
+			wlc_view_bring_to_front(*handle);
+		}
```

Comment:
Does this transfer ownership of unmanaged containers to outputs instead of workspaces? That doesn't make sense. Try this: run dmenu and switch workspaces. If dmenu shows up behind the windows on the new workspace, bug.